### PR TITLE
Adding Rubygem for Ruby Applications

### PR DESCRIPTION
**Pull request by christopherhein**
**Why:**

* Currently tested to run a `sinatra` application
* Uses the same `mp-compiler` to build out the specs
* Uses an `alpine` image out of the box to run

**Todo:**
- [x] need to set sane defaults in the `gem` to allow you to run without ports defined
- [x] build out sharding tutorial

Resolves https://github.com/metaparticle-io/package/issues/8

### CODE REVIEW

File: ruby/metaparticle/CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# Contributor Covenant Code of Conduct
```
Comment (brendandburns) [inline ruby/metaparticle/CODE_OF_CONDUCT.md]: I'd like to add a code of conduct to the overall repository, and I'd like to use the Microsoft CoC:

https://opensource.microsoft.com/codeofconduct/

Let's do this as a separate PR?

File: ruby/metaparticle/LICENSE.txt [hunk 2]
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
```
Comment (brendandburns) [inline ruby/metaparticle/LICENSE.txt]: I'd like to have a single LICENSE at the top, rather than a lower-level one in a language implemenation. Hope that makes sense...

File: ruby/metaparticle/README.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+## Development
+
+After checking out the repo, run `bin/setup` to install dependencies. Then, run `rake spec` to run the tests. You can also run `bin/console` for an interactive prompt that will allow you to experiment.
+
+To install this gem onto your local machine, run `bundle exec rake install`. To release a new version, update the version number in `version.rb`, and then run `bundle exec rake release`, which will create a git tag for the version, push git commits and tags, and push the `.gem` file to [rubygems.org](https://rubygems.org).
+
+## Contributing
+
+Bug reports and pull requests are welcome on GitHub at https://github.com/christopherhein/metaparticle. This project is intended to be a safe, welcoming space for collaboration, and contributors are expected to adhere to the [Contributor Covenant](http://contributor-covenant.org) code of conduct.
```
Comment (brendandburns) [inline ruby/metaparticle/README.md]: Change to point to the https://github.com/metaparticle-io/package repo.

Delete the code of conduct part given we'll add it in a separate PR for the whole repo.

File: ruby/metaparticle/README.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+## License
+
+The gem is available as open source under the terms of the [MIT License](http://opensource.org/licenses/MIT).
+
+## Code of Conduct
+
+Everyone interacting in the Metaparticle project’s codebases, issue trackers, chat rooms and mailing lists is expected to follow the [code of conduct](https://github.com/christopherhein/metaparticle/blob/master/CODE_OF_CONDUCT.md).
```
Comment (brendandburns) [inline ruby/metaparticle/README.md]: Let's delete and add these as part of the Code of Conduct PR (or wait to merge this, and redirect it)

**Comment by brendandburns** [comment]
@christopherhein thanks for the PR, I had a few comments...

**Comment by christopherhein** [comment]
@brendandburns addressed the current feedback, thank you!

**Comment by brendandburns** [comment]
LGTM thanks!
